addons/survey/models/survey.py

```diff
@@ -0,0 +1,1087 @@
+        old_input_line = self.search([('user_input_id', '=', user_input_id),
+                                        ('survey_id', '=', question.survey_id.id),
+                                        ('question_id', '=', question.id)], limit=1)
+        if old_input_line:
+            old_input_line.write(vals)
+        else:
+            self.create(vals)
+        return True
+
+    @api.model
+    def save_line_textbox(self, user_input_id, question, post, answer_tag):
+        vals = {
+            'user_input_id': user_input_id,
+            'question_id': question.id,
+            'skipped': False
+        }
+        if answer_tag in post and post[answer_tag].strip() != '':
+            vals.update({'answer_type': 'text', 'value_text': post[answer_tag]})
+        else:
+            vals.update({'answer_type': None, 'skipped': True})
+        old_input_line = self.search([('user_input_id', '=', user_input_id),
+                                        ('survey_id', '=', question.survey_id.id),
+                                        ('question_id', '=', question.id)], limit=1)
+        if old_input_line:
+            old_input_line.write(vals)
+        else:
+            self.create(vals)
+        return True
+
+    @api.model
+    def save_line_numerical_box(self, user_input_id, question, post, answer_tag):
+        vals = {
+            'user_input_id': user_input_id,
+            'question_id': question.id,
+            'skipped': False
+        }
+        if answer_tag in post and post[answer_tag].strip() != '':
+            vals.update({'answer_type': 'number', 'value_number': float(post[answer_tag])})
+        else:
+            vals.update({'answer_type': None, 'skipped': True})
+        old_input_line = self.search([('user_input_id', '=', user_input_id),
+                                        ('survey_id', '=', question.survey_id.id),
+                                        ('question_id', '=', question.id)], limit=1)
+        if old_input_line:
+            old_input_line.write(vals)
+        else:
+            self.create(vals)
+        return True
+
+    @api.model
+    def save_line_datetime(self, user_input_id, question, post, answer_tag):
+        vals = {
+            'user_input_id': user_input_id,
+            'question_id': question.id,
+            'skipped': False
+        }
+        if answer_tag in post and post[answer_tag].strip() != '':
+            vals.update({'answer_type': 'date', 'value_date': post[answer_tag]})
+        else:
+            vals.update({'answer_type': None, 'skipped': True})
+        old_input_line = self.search([('user_input_id', '=', user_input_id),
+                                        ('survey_id', '=', question.survey_id.id),
+                                        ('question_id', '=', question.id)], limit=1)
+        if old_input_line:
+            old_input_line.write(vals)
+        else:
+            self.create(vals)
+        return True
```

Comment:
Up to this, all "save_line_+question_type" having almost same stuff, can't we create one generic method for this common stuff(It will be considered part of refactoring so its should not be on priority)